doc/docusaurus/docs/working-with-scripts/other-optimization-techniques.md

```diff
@@ -128,6 +128,37 @@ Traces can be expensive especially in terms of script sizes.
 It is advisable to use traces during development, but to remove them when deploying your scripts on mainnet.
 Traces can be removed via the `remove-trace` plugin flag.
 
+## Using `BuiltinArray` for index-based lookups
+
+For multiple lookups by index, converting a list to a `BuiltinArray` can be more cost-effective than performing lookups on the list directly. The initial cost of converting the list to an array can be offset by the significantly faster lookups.
+
+As a rule of thumb, if you only need to perform a single lookup, it is likely not worth the conversion cost. The benefits of using a `BuiltinArray` become apparent when you need to perform several lookups on the same data structure.
```

Comment:
> it is likely not worth the conversion cost.

What about picking list vs. array to use in the first place? If you use array in your data type in the first place, then you won't need the conversion. So should one generally prefer using array or list in the first place, and why?